### PR TITLE
feature/direct-to-pdf-availability-check-BZ-4508

### PR DESCRIPTION
**Pull request by tiandavis**
## Summary - [BZ-4508](https://thirdiron.atlassian.net/browse/BZ-4508)

Checks article is available before display Download to PDF link; Uses Font Awesome PDF document icon; Drops platform prefixes from customer facing configs.

## Deploy Precautions

- None

### CODE REVIEW

File: src/primo/browzine-primo-adapter.js [hunk 1]
```diff
@@ -214,12 +214,12 @@ browzine.primo = (function() {
   };
 
   function directToPDFTemplate(directToPDFUrl) {
-    var pdfIcon = "https://s3.amazonaws.com/thirdiron-assets/images/integrations/browzine-pdf-download-icon.png";
-    var articlePDFDownloadLinkText = browzine.primoArticlePDFDownloadLinkText || "Download Now";
+    var pdfIcon = "https://s3.amazonaws.com/thirdiron-assets/images/integrations/browzine-pdf-download-icon.svg";
```
Comment (KeeganJ) [inline src/primo/browzine-primo-adapter.js]: Nice switch to svg!

File: README.md [hunk 2]
```diff
@@ -63,13 +63,13 @@ e.g. You can customize "View the Journal", "View Complete Issue", "Browse Now" a
...
-  summonArticlePDFDownloadLinkEnabled: true,
-  summonArticlePDFDownloadWording: "Article PDF",
-  summonArticlePDFDownloadLinkText: "Download Now",
+  journalWording: "View the Journal",
+  articleWording: "View Complete Issue",
+  journalBrowZineWebLinkText: "Browse Now",
+  articleBrowZineWebLinkText: "Browse Now",
+  articlePDFDownloadLinkEnabled: true,
+  articlePDFDownloadWording: "Article PDF",
+  articlePDFDownloadLinkText: "Download Now",
```
Comment (KeeganJ) [inline README.md]: I like the rename, much easier to read!

File: tests/acceptance/browzine-primo-adapter-test.js [hunk 3]
```diff
@@ -301,6 +301,88 @@ describe("BrowZine Primo Adapter >", function() {
...
+      });
+
+      it("should not have an enhanced browse article in browzine link in search result", function() {
+        var template = searchResult.find(".browzine .browzine-web-link");
+        expect(template.text().trim()).toEqual("");
+      });
+
+      it("should not have an enhanced direct to pdf link in search result", function() {
+        var template = searchResult.find(".browzine .browzine-direct-to-pdf-link");
+        expect(template.text().trim()).toEqual("");
```
[the remark this answers is not in the document]
Comment (tiandavis) [reply tests/acceptance/browzine-primo-adapter-test.js]: They've really helped a lot.

File: src/primo/browzine-primo-adapter.js [hunk 4]
```diff
@@ -313,7 +314,7 @@ browzine.primo = (function() {
           })();
         }
 
-        if(browzineWebLink) {
+        if(browzineWebLink && browzineEnabled) {
```
[the remark this answers is not in the document]
Comment (tiandavis) [reply src/primo/browzine-primo-adapter.js]: > Is this here mostly to make it more bulletproof in the future?

Yep. Bulletproof it.

> Is this here mostly to make it more bulletproof in the future? Or is there some examples where we'd have a browzineWebLink but a journal wouldn't be browzineEnabled?

Not that I know of as of today.

Of course, if we remove the check, that may help us catch any potential API issues like it did with the bug Sarah found...

I could go either way.

**Comment by karlbecker** [comment]
Hey @tiandavis ,
Aha, I can help.  I think if you wait long enough, it would bust itself. 

However, if you go [into CloudFront](
https://console.aws.amazon.com/cloudfront/home?region=us-east-1#) and go to the CloudFront distribution for assets.thirdiron.com , looks like you can Invalidate a file as a one-time operation. I'm looking [at these docs](
https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/Invalidation.htm).

I just tried an invalidation, and it took a minute or so to finish the operation, but now it seems right - check out the proper image now that I see!

screen shot 2018-08-14 at 5 30 00 pm

**Comment by tiandavis** [comment]
Wow! Adding to my notes! Thanks @karlbecker!

**Pull request merged**